.strict-typing

```diff
@@ -87,6 +87,7 @@ homeassistant.components.group.*
 homeassistant.components.guardian.*
 homeassistant.components.history.*
 homeassistant.components.homeassistant.triggers.event
+homeassistant.components.homekit_controller
```

Comment:
What is the difference between:
- `homeassistant.components.homekit_controller`
and
- `homeassistant.components.homekit_controller.*`

Reply:
Good spot. I didn't even see all the `*` somehow and I didn't notice the mistake because it triggered a bunch more errors for me to clean up in multiple files. Boo. I think this just made `__init__.py` strict. BRB...

Reply:
Yep, that only covers __init__.py. I think the extra errors in climate.py etc were because of enabling the type hints in aiohomekit to be used. 

2 choices:

* Add the files that do pass with strict typing (e.g. storage.py does) individually (recorder seems to do this)
* Back out the change to `.strict-typing` but still submit these typing fixes as they are more correct.

I've done the second for now. Do you have a preference?

Reply:
I would say that adding individual files is better than not adding anything to` .strict-typing`.
It is similar to #65416 where I think excluding individual files is better than excluding whole components.

Reply:
OK - last commit does that. Was able to enable for 7 homekit_controller modules.